suppressible-error-prone/src/main/java/com/palantir/suppressibleerrorprone/ReportedFixCache.java

```diff
@@ -0,0 +1,104 @@
+    public static final Predicate<String> KEEP_EVERYTHING = bc -> true;
+
+    ReportedFixCache() {}
+
+    /**
+     * When first called on a {@code declaration}, initialize a {@code LazySuppressionFix} on it, choosing which
+     * suppressions to keep with {@code filterExisting}. Subsequent calls on this {@code declaration} will return the
+     * initialized {@code LazySuppressionFix}.
+     */
+    public LazySuppressionFix getOrReportNew(
+            TreePath declaration, VisitorState visitorState, Predicate<String> filterExisting) {
+        return cache.computeIfAbsent(
+                declaration.getLeaf(), _ignored -> createAndReportFix(declaration, visitorState, filterExisting));
+    }
+
+    /**
+     * Initialize a {@code LazySuppressionFix} on {@code declaration}, choosing which suppressions to keep with
+     * {@code filterExisting}.
+     */
+    @SuppressWarnings("PreferSafeLoggableExceptions") // It doesn't matter in our internal codebases
+    public LazySuppressionFix reportNew(
+            TreePath declaration, VisitorState visitorState, Predicate<String> filterExisting) {
+        if (cache.containsKey(declaration.getLeaf())) {
+            throw new IllegalArgumentException("A fix on this declaration already exists");
+        }
+        return cache.put(declaration.getLeaf(), createAndReportFix(declaration, visitorState, filterExisting));
+    }
+
+    private LazySuppressionFix createAndReportFix(
+            TreePath declaration, VisitorState state, Predicate<String> filterExisting) {
+        if (!SuppressWarningsUtils.suppressibleTreePath(declaration)) {
+            throw new IllegalArgumentException("Not suppressible: " + declaration);
+        }
+        ModifiersTree modifiersTree = AnnotationUtils.getModifiers(declaration.getLeaf());
+        Optional<? extends AnnotationTree> suppressWarnings = modifiersTree.getAnnotations().stream()
+                .filter(annotation -> {
+                    Name annotationName = AnnotationUtils.annotationName(annotation.getAnnotationType());
+                    return annotationName.contentEquals(CommonConstants.SUPPRESS_WARNINGS_ANNOTATION);
+                })
+                .findFirst();
+
+        Stream<String> allExistingSuppressions =
+                suppressWarnings.map(AnnotationUtils::annotationStringValues).orElseGet(Stream::of);
+        Set<String> filteredExistingSuppressions =
+                allExistingSuppressions.filter(filterExisting).collect(Collectors.toSet());
+
+        LazySuppressionFix fix = new LazySuppressionFix(
+                Optional.ofNullable(state.getSourceCode()),
+                suppressWarnings,
+                declaration.getLeaf(),
+                filteredExistingSuppressions);
+        @SuppressWarnings("RestrictedApi")
+        Description description = Description.builder(
+                        (DiagnosticPosition) declaration.getLeaf(),
+                        "SuppressibleErrorProne",
```

Comment:
To make it really obvious that it's the bug checker in this repo, could do `SuppressibleErrorProne.class.getSimpleName()`.